ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/ModuleImporter.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell;
+
+import io.ballerina.projects.Package;
+import io.ballerina.projects.PackageDependencyScope;
+import io.ballerina.projects.PackageDescriptor;
+import io.ballerina.projects.PackageName;
+import io.ballerina.projects.PackageOrg;
+import io.ballerina.projects.PackageVersion;
+import io.ballerina.projects.environment.PackageRepository;
+import io.ballerina.projects.environment.ResolutionOptions;
+import io.ballerina.projects.environment.ResolutionRequest;
+import io.ballerina.projects.internal.environment.BallerinaDistribution;
+import io.ballerina.projects.internal.environment.DefaultEnvironment;
+import org.wso2.ballerinalang.compiler.util.Names;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Support class to add required import statements.
+ *
+ * @since 2.0.0
+ */
+public class ModuleImporter {
+
+    private final List<Package> packageList;
+    private static final String UNDEFINED_MODULE = "undefined module";
+    private static final String LANG = "lang";
+    private static final List<String> SKIPPED_LIBS = Arrays.asList("lang.annotations", "lang.__internal", "lang.query");
+
+    public ModuleImporter() {
+        packageList = getPackagesFromDistRepo();
+    }
+
+    /**
+     * Check module availability in the distribution.
+     *
+     * @param module module name.
+     * @return availability of the module.
+     */
+    public boolean isModuleInDistRepo(String module) {
+        boolean isModuleInDistRepo = false;
+        String langModule = LANG + "." + module;
+        for (Package pkg : packageList) {
+            if (module.equals(pkg.packageName().value()) || langModule.equals(pkg.packageName().value())) {
```

Comment:
It is not mandatory to import lang modules, right? Then why are we covering this case?

Reply:
if we consider this example https://ballerina.io/learn/by-example/json-type user should be notified to import related Lang module.